### PR TITLE
add AppBio Quantstudio Design and Analysis adapter

### PR DESCRIPTION
**Pull request by alejandro-salgado**
- Add AppBio Quantstudio Design and Analysis adapter

### CODE REVIEW

File: src/allotropy/parsers/appbio_quantstudio_designandanalysis/appbio_quantstudio_designandanalysis_contents.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+from allotropy.parsers.utils.values import (
+    assert_not_empty_df,
+    assert_not_none,
+)
+from allotropy.types import IOType
+
+
+class DesignQuantstudioContents:
+    def __init__(self, contents: IOType) -> None:
+        self.raw_contents = pd.read_excel(contents, header=None, sheet_name=None)
```
Comment (nathan-stender) [inline src/allotropy/parsers/appbio_quantstudio_designandanalysis/appbio_quantstudio_designandanalysis_contents.py:17]: contents and raw_contents are not accessed after they are used to create header and data, they should not be stored as member variables. Instead they should be normal variables, and should be passed into _get_header and _get_data

File: src/allotropy/parsers/appbio_quantstudio_designandanalysis/appbio_quantstudio_designandanalysis_structure.py [hunk 2]
```diff
@@ -0,0 +1,569 @@
...
+            ),
+            model_number=try_str_from_series_or_none(header, "Instrument Type") or "NA",
+            device_serial_number=(
+                try_str_from_series_or_none(header, "Instrument Serial Number") or "NA"
+            ),
+            measurement_method_identifier=try_str_from_series(
+                header, "Quantification Cycle Method"
+            ),
+            pcr_detection_chemistry=(
+                try_str_from_series_or_none(header, "Chemistry") or "N/A"
```
Comment (nathan-stender) [inline src/allotropy/parsers/appbio_quantstudio_designandanalysis/appbio_quantstudio_designandanalysis_structure.py:109]: We are using "NA" above, but "N/A" here, which is correct?

File: src/allotropy/parsers/utils/values.py [hunk 3]
```diff
@@ -183,7 +191,7 @@ def try_bool_from_series_or_none(
 ) -> Optional[bool]:
     try:
         value = data.get(key)
-        return None if value is None else bool(value)
+        return None if value is None else str(value).lower() == "true"
```
Comment (nathan-stender) [inline src/allotropy/parsers/utils/values.py:194]: Ah, I should have caught this originally.  Can you add a function like this:

```
def str_to_bool(v):
  return v.lower() in ("yes", "true", "t", "1")
``` 

And call that instead?

```
return None if value is None else str_to_bool(value)
```

File: tests/parsers/appbio_quantstudio_designandanalysis/appbio_quantstudio_designandanalysis_parser_test.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    "appbio_quantstudio_designandanalysis_QS7_Standard_Curve_example06",
+    "appbio_quantstudio_designandanalysis_QS7Pro_Genotyping_example07",
+    "appbio_quantstudio_designandanalysis_QS7Pro_Multiplex_example08",
+    "appbio_quantstudio_designandanalysis_QS7Pro_PCR_with_Melt_example09",
+    "appbio_quantstudio_designandanalysis_QS7Pro_Presence_and_Absence_example10",
+    "appbio_quantstudio_designandanalysis_QS7Pro_Relative_Quantification_example11",
+    "appbio_quantstudio_designandanalysis_QS7Pro_Relative_Quantification_Biogroup_example12",
+    "appbio_quantstudio_designandanalysis_QS7Pro_Relative_Standard_Curve_example13",
+    "appbio_quantstudio_designandanalysis_QS7Pro_Standard_Curve_example14",
+    ## "appbio_quantstudio_designandanalysis_QS7Pro_Standard_Curve_TAC_example15",
```
Comment (nathan-stender) [inline tests/parsers/appbio_quantstudio_designandanalysis/appbio_quantstudio_designandanalysis_parser_test.py:21]: Add comment on why these tests are excluded, or remove?

**Review by nathan-stender — CHANGES_REQUESTED** [review]
Great PR! A couple changes

File: tests/parsers/appbio_quantstudio_designandanalysis/appbio_quantstudio_designandanalysis_parser_test.py (hunk 4 above)
Comment (alejandro-salgado) [reply tests/parsers/appbio_quantstudio_designandanalysis/appbio_quantstudio_designandanalysis_parser_test.py:21]: Those are files not supported in this version, but they should be supported in the next one

**Review by nathan-stender — APPROVED** [review]

**Review by nathan-stender — APPROVED** [review]

**Pull request merged**